### PR TITLE
Work around case where no author is in the bibtex entry.

### LINKED ISSUE #9154: Documentation references can't handle no author
### Description of the enhancement or error report
If a bibtex entry doesn't have an author, the documentation system dies.  This is true even if the bibtex entry does not require an author.

### Rationale for the enhancement or information for reproducing the error
Allow the documentation to be built even if an author was not listed.

### Identified impact
None.

I'll put a pull request in shortly.

### PR DESCRIPTION
**Pull request by jasondhales**
### Complete each of the following items before creating a PR
Closes #9154 

@tolman42

### CODE REVIEW

**Comment by aeslaughter** [comment]
I would like a test for this before it is merged. The test should be added to python/MooseDocs/tests/bibtex/test_bibtex.py

**Comment by jasondhales** [comment]
I'm happy to add a test.  How do I run the doc tests?

**Comment by aeslaughter** [comment]
```
cd moose/python
./run_tests -j12 --re MooseDocs
```

File: python/MooseDocs/tests/bibtex/gold/test_no_author.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<p><span data-moose-cite="\cite{noAuthorTestKey}"><a href="#noAuthorTestKey">ABC (2017)</a></span>\n<ol class="moose-bibliography" data-moose-bibfiles="[u'/Users/halejd/gitProjects/moose/docs/bib/test.bib']">
```
Comment (aeslaughter) [inline python/MooseDocs/tests/bibtex/gold/test_no_author.html]: {{ROOT_DIR}}//moose/docs/bib/test.bib

**Review by aeslaughter — CHANGES_REQUESTED** [review]

File: python/MooseDocs/tests/bibtex/gold/test_no_author.html (hunk 1 above)
Comment (jasondhales) [inline python/MooseDocs/tests/bibtex/gold/test_no_author.html]: All the others used CWD, and so I changed it to that.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**